### ISSUE TITLE
Math Expressions

### ISSUE THREAD

**Issue by svickers** [nebrelbug/squirrelly.js.org#5]
Is there any way to do expressions in squirrelly syntax?  I get javascript errors when something like this is attempted:
`{{obj.thirdchild[1]-obj.thirdchild[2]}}`

However, this works fine:
`{{obj.thirdchild[1]-2}}`

**Comment by bgub** [comment]
Yep! You’ll need to prefix data references other than the first with `options` though, so

```
{{obj.thirdchild[1]-options.obj.thirdchild[2]}}
```
Let me know if this helps!

**Comment by bgub** [comment]
Closing this now :)

**Comment by svickers** [comment]
Thanks that helps.  I'm a little confused on when to use options and when not to, is there guidance around what it is?

**Comment by bgub** [comment]
@svickers behind the scenes, Squirrelly turns your template into a function. Here's a simplified example:

```
Cool template, {{data}}
```
Could get turned into:
```js
function template(options) {
var result = "Cool template, "
result += options.data
return result
}
```
But Squirrelly only puts `options` at the beginning of each reference, so if you have:
```
Cool template, {{data + otherData}}
```
Squirrelly will treat `otherData` like a global variable.
```js
function template(options) {
var result = "Cool template, "
result += options.data + otherData
return result
}
```
**TL;DR**
You need to put `options` before a reference that's not the very first item inside `{{ ... }}`.

Does that help?

**Comment by svickers** [comment]
Makes sense, thanks!
